daemon/api.go

```diff
@@ -2347,12 +2347,9 @@ func runSnapctl(c *Command, r *http.Request, user *auth.UserState) Response {
 		return BadRequest("snapctl cannot run without args")
 	}
 
-	// Right now snapctl is only used for hooks. If at some point it grows
-	// beyond that, this probably shouldn't go straight to the HookManager.
-	context, err := c.d.overlord.HookManager().Context(snapctlOptions.ContextID)
-	if err != nil {
-		return BadRequest("error running snapctl: %s", err)
-	}
+	// Ignore missing context error to allow 'snapctl -h' without a context;
+	// Actual context is validated later by get/set.
+	context, _ := c.d.overlord.HookManager().Context(snapctlOptions.ContextID)
 	stdout, stderr, err := ctlcmd.Run(context, snapctlOptions.Args)
```

Comment:
This line would never get a nil context, but now it can. Are we sure it won't crash (panic) if the context is actually used (say, "snapctl get" out of hooks, rather than -h).

Reply:
Yes, it's fine, we check for nils early in get/set and we have unit tests for nil context for both get and set.